src/connectors/impls/gbq/writer/sink.rs

```diff
@@ -0,0 +1,660 @@
+            | TableType::Timestamp => field_descriptor_proto::Type::String,
+            TableType::Struct => {
+                let type_name_for_field = format!("struct_{}", raw_field.name);
+                let mapped = map_field(&type_name_for_field, &raw_field.fields);
+                nested_types.push(mapped.0);
+                subfields = mapped.1;
+
+                type_name = Some(type_name_for_field);
+                field_descriptor_proto::Type::Message
+            }
+
+            TableType::Unspecified => {
+                warn!("Found a field of unspecified type: {}", raw_field.name);
+                continue;
+            }
+        };
+
+        proto_fields.push(FieldDescriptorProto {
+            name: Some(raw_field.name.to_string()),
+            number: Some(i32::from(tag)),
+            label: None,
+            r#type: Some(i32::from(grpc_type)),
+            type_name,
+            extendee: None,
+            default_value: None,
+            oneof_index: None,
+            json_name: None,
+            options: None,
+            proto3_optional: None,
+        });
+
+        fields.insert(
+            raw_field.name.to_string(),
+            Field {
+                table_type,
+                tag: u32::from(tag),
+                subfields,
+            },
+        );
+
+        tag += 1;
+    }
+
+    (
+        DescriptorProto {
+            name: Some(schema_name.to_string()),
+            field: proto_fields,
+            extension: vec![],
+            nested_type: nested_types,
+            enum_type: vec![],
+            extension_range: vec![],
+            oneof_decl: vec![],
+            options: None,
+            reserved_range: vec![],
+            reserved_name: vec![],
+        },
+        fields,
+    )
+}
+
+fn encode_field(val: &Value, field: &Field, result: &mut Vec<u8>) -> Result<()> {
+    let tag = field.tag;
+
+    // fixme check which fields are required and fail if they're missing
+    // fixme do not panic if the tremor type does not match
+    match field.table_type {
+        TableType::Double => prost::encoding::double::encode(
+            tag,
+            &val.as_f64()
+                .ok_or_else(|| ErrorKind::BigQueryTypeMismatch("f64", val.value_type()))?,
+            result,
+        ),
+        TableType::Int64 => prost::encoding::int64::encode(
+            tag,
+            &val.as_i64()
+                .ok_or_else(|| ErrorKind::BigQueryTypeMismatch("i64", val.value_type()))?,
+            result,
+        ),
+        TableType::Bool => prost::encoding::bool::encode(
+            tag,
+            &val.as_bool()
+                .ok_or_else(|| ErrorKind::BigQueryTypeMismatch("bool", val.value_type()))?,
+            result,
+        ),
+        TableType::String
+        | TableType::Date
+        | TableType::Time
+        | TableType::Datetime
+        | TableType::Timestamp
+        | TableType::Numeric
```

Comment:
shouldn't we also allow any number type for `Numeric` and `Bignumeric` too?

Reply:
`Numeric` and `Bignumeric` are decimal-precision, so I'd avoid doing that, because if somebody tries to push a float/double into them, it's likely a (potentially costly) mistake.

Reply:
Add a comment to this effect with the reasoning here would be a good change